x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtRealm.java

```diff
@@ -459,20 +485,34 @@ private String buildTokenPrincipal(JWTClaimsSet jwtClaimsSet) {
         final FallbackableClaim subClaim = new FallbackableClaim("sub", fallbackClaimNames, jwtClaimsSet);
         final String subject = subClaim.getStringClaimValue();
         if (false == Strings.hasText(subject)) {
-            logger.debug("claim [{}] is missing for building token principal for realm [{}]", subClaim, name());
+            logger.debug(
+                "Can not build token principal for realm [{}] since claim [{}] is missing from claimSet [{}]",
+                name(),
+                subClaim,
+                jwtClaimsSet
+            );
```

Comment:
Similarly, can we wrap it or use supplier for the claimSet?

Reply:
I don't think this one is an issue since we already paid the cost to get a reference to the claimset (which requires creating and parsing a json objects) and the toString is not evaluated until logged.

Reply:
You are right. I mis-read the code.